### ISSUE TITLE
Speed up sigmoid in cases of non 32x32 tiles.

### ISSUE THREAD

**Issue by pavlepopovic** [tenstorrent/tt-metal#19477]
**Describe the bug**
In a customer model, we have a convolution which outputs a tensor of logical shape [1, 1, 2048*128, 2]. Due to convolution/tt-metal op constraints, this shape will actually get padded to [1, 1, 2048 * 128, 32].
Following the convolution, sigmoid is called.
Sigmoid will operate on a tile-by-tile basis, calculating the results even for the padded last dim.
In this case, the duration of the sigmoid will be somewhere around 220k cycles, amounting for more than 25% device perf of the full model.

Now, ideally, we would not be computing sigmoid for full tiles (32x2 is what is needed).
Sigmoid LLK already has a mechanism which somewhat addresses this, and it's called VectorMode.
Using VectorMode::C (which tells SFPU to ignore faces 1 & 4) drops duration to ~120k cycles, almost a 2x speedup :)

To do that, I've hacked the sigmoid_tile HLK call to always use VectorMode::C, like this:
`ALWI void sigmoid_tile(uint32_t idst) { MATH((llk_math_eltwise_unary_sfpu_sigmoid<APPROX>(idst, 2))); }
`
So, as a first step of optimising sigmoid (and other unary ops will benefit from this, when needed), I think exposing VectorMode on unary ops should be a good start.

Ideally, we would want to perform computation on only 32x2 actual data, which LLK doesn't support ATM.

**To Reproduce**
Attaching a unit test:

```
import pytest

import torch

import ttnn

from tests.ttnn.utils_for_testing import assert_with_pcc, assert_equal
from tests.ttnn.unit_tests.operations.eltwise.backward.utility_funcs import data_gen_with_range, compare_pcc
from models.utility_functions import torch_random, skip_for_grayskull, is_wormhole_b0, is_blackhole


def run_unary_test_sharded(device, hw, out_channels, ttnn_function, pcc=0.9999):
    torch.manual_seed(0)

    torch_input_tensor = torch.rand((hw, out_channels), dtype=torch.bfloat16)
    golden_function = ttnn.get_golden_function(ttnn_function)
    torch_output_tensor = golden_function(torch_input_tensor, device=device)

    num_cores = 64
    core_range_set = ttnn.num_cores_to_corerangeset(num_cores, device.compute_with_storage_grid_size())
    shard_shape = [
        hw // num_cores,
        out_channels if out_channels >= 32 else 32
    ]

    memory_config = ttnn.MemoryConfig(
                ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
                ttnn.BufferType.L1,
                ttnn.ShardSpec(
                    core_range_set,
                    shard_shape,
                    ttnn.ShardOrientation.ROW_MAJOR,
                ),
            )

    input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, device=device, memory_config=memory_config)
    output_tensor = ttnn_function(input_tensor)
    output_tensor = ttnn.from_device(output_tensor)
    output_tensor = ttnn.to_torch(output_tensor)

    assert_with_pcc(torch_output_tensor, output_tensor, pcc)

@pytest.mark.parametrize("h", [2048])
@pytest.mark.parametrize("w", [128])
@pytest.mark.parametrize("out_channels", [2])
def test_sigmoid_two_out_channels(device, h, w, out_channels):
    torch.manual_seed(0)
    run_unary_test_sharded(device, h * w, out_channels, ttnn.sigmoid, pcc=0.991)

```
To try VectorMode::C, go to tt_metal/include/compute_kernel_api.h, and modify the implementation to be like this;

<img width="960" alt="Image" src="https://github.com/user-attachments/assets/83871f51-3df4-47f2-b7de-8b0dab6c8e35" />

That should give a speedup, and maintain PCC accuracy.

**Comment by pavlepopovic** [comment]
@mbezuljTT fyi

**Comment by pavlepopovic** [comment]
I've also noticed one more thing:
- There is a sigmoid approx mode implementation in the repo itself. It gives us a ~5x speedup, combined with vector mode 2x speedup, it should be fine. (PCC is okay)
- There is no way to access this function, as llk_math_eltwise_unary_sfpu_sigmoid_init/llk_math_eltwise_unary_sfpu_sigmoid calls templated with APPROX=True don't forward to this function, but rather just call the non-approx mode version of it. The correct functions that should be called in this case is: llk_math_eltwise_unary_sfpu_sigmoid_appx_init/llk_math_eltwise_unary_sfpu_sigmoid_appx
- There is no way to pass in math_approx_mode=True to a Unary op :(

So we need at least 2 fixes to get this working - expose APPROX mode on unary op, and expose correct function in this case

**Comment by umadevimcw** [comment]
@pavlepopovic  Will update and submit a PR

**Comment by umadevimcw** [comment]
@cmaryanTT  @pavlepopovic  has started to work on this. So to avoid duplication i will be working on other issue and removing myself from assignee.

**Comment by pavlepopovic** [comment]
@cmaryanTT  I've opened the PR for approx_mode exposing: https://github.com/tenstorrent/tt-metal/pull/19768
@umadevimcw I haven't been experimentign with exposing vector mode on this op, and don't have anytihng for it yet. Can you take on that following the merge of the PR?
